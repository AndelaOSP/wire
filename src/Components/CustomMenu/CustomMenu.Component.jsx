import React from 'react';
import PropTypes from 'prop-types';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';

const styles = {
  customWidth: {
    width: 145,
  },
};

/**
 * @class CustomMenu
 */
export class CustomMenu extends React.Component {
  constructor() {
    super();

    this.state = {
      value: 'All Countries',
    };
  }

  /**
   * Method to handle menu item selection
   */
  handleChange = (event, index, value) => this.props.changeCountryFilter(value);

  render() {
    return (
      <div>
        <SelectField
          value={this.state.value}
          onChange={this.handleChange}
          style={styles.customWidth}
        >
          <MenuItem value="All Countries" primaryText="All Countries" />
          <MenuItem value="Nairobi, Kenya" primaryText="Kenya" />
          <MenuItem value="Lagos, Nigeria" primaryText="Nigeria" />
          <MenuItem value="Kampala, Uganda" primaryText="Uganda" />
          <MenuItem value="New York, USA" primaryText="USA" />
        </SelectField>
      </div>
    );
  }
}

CustomMenu.propTypes = {
  changeCountryFilter: PropTypes.func,
};

export default CustomMenu;
