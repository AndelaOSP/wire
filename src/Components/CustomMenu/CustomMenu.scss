@import '../../theme.scss';

.country-menu {
    // background-color: map-get($colors, btn-background-inactive) !important;
    // border: 0.09rem solid map-get($colors, btn-border-inactive) !important;
    // border-radius: 6rem !important;
    // max-height: 1.7rem !important;
}