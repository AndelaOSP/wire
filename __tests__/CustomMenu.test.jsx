import React from 'react';
import shallowToJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';
import CustomMenu from '../src/Components/CustomMenu/CustomMenu.Component';
import SelectField from 'material-ui/SelectField';

describe('CustomMenu component', () => {
  it('Should have all the CustomMenu content', () => {
    const customMenu = shallow(<CustomMenu />);
    const tree = shallowToJSON(customMenu);
    expect(tree).toMatchSnapshot();
  });

  it('Should change state when filter value changes', () => {
    const changeCountryFilterMock = jest.fn()
    const menu = shallow(<CustomMenu changeCountryFilter={changeCountryFilterMock}/>)
    const select = menu.find(SelectField);
    select.props().onChange()
    menu.setState({value: 'Kenya'})
    menu.update()
    expect(menu.instance().state.value).toEqual('Kenya');
    expect(changeCountryFilterMock.mock.calls.length).toBe(1);
  });
});
