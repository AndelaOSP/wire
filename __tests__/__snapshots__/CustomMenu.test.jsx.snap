// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomMenu component Should have all the CustomMenu content 1`] = `
<div>
  <SelectField
    autoWidth={false}
    disabled={false}
    fullWidth={false}
    multiple={false}
    onChange={[Function]}
    style={
      Object {
        "width": 145,
      }
    }
    value="All Countries"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="All Countries"
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
      value="All Countries"
    />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Kenya"
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
      value="Nairobi, Kenya"
    />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Nigeria"
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
      value="Lagos, Nigeria"
    />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Uganda"
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
      value="Kampala, Uganda"
    />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="USA"
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
      value="New York, USA"
    />
  </SelectField>
</div>
`;
